cli/flox/src/commands/environment.rs

```diff
@@ -1231,9 +1252,29 @@ impl Uninstall {
             self.packages.as_slice().join(", "),
             self.environment
         );
-        let concrete_environment = self
+        let concrete_environment = match self
             .environment
-            .detect_concrete_environment(&flox, "uninstall from")?;
+            .detect_concrete_environment(&flox, "uninstall from")
+        {
+            Ok(concrete_environment) => concrete_environment,
+            Err(EnvironmentSelectError::Environment(
+                ref e @ EnvironmentError2::DotFloxNotFound(ref dir),
+            )) => {
+                bail!(formatdoc! {"
+                {e}
+
+                Create an environment with 'flox init --dir {}'", dir.to_string_lossy()
+                })
+            },
+            Err(e @ EnvironmentSelectError::EnvNotFoundInCurrentDirectory) => {
+                bail!(formatdoc! {"
+                {e}
+
+                Create an environment with 'flox init' or uninstall packages from an environment found elsewhere with 'flox uninstall {} --dir <path>'",
+                self.packages.join(" ")})
+            },
+            Err(e) => Err(e)?,
+        };
```

Comment:
comment (nb): lets just keep an eye on this whether we end up needing this kind of messaging for more/all commands. Two almost copies are prob fine, but we should consider factoring this out if it becomes more

Reply:
Yeah the reason I copied it is because the wording is slightly different (one has `install`, one has `uninstall`). But yeah we should definitely keep an eye on it